Sources/PointFree/Account/Change.swift

```diff
@@ -0,0 +1,392 @@
+}
+
+private func requireValidSeating(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, Tuple4<Stripe.Subscription, Database.User, Int, Pricing>, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, Tuple4<Stripe.Subscription, Database.User, Int, Pricing>, Data> {
+
+    return filter(
+      seatsAvailable,
+      or: redirect(
+        to: .account(.index),
+        headersMiddleware: flash(
+          .error,
+          "We can’t reduce the number of seats below the number that are active."
+        )
+      )
+      )
+      <| middleware
+}
+
+private func seatsAvailable(_ data: Tuple4<Stripe.Subscription, Database.User, Int, Pricing>) -> Bool {
+  let (_, _, seatsTaken, pricing) = lower(data)
+
+  return pricing.quantity >= seatsTaken
+}
+
+private let extraStyles =
+  ((input & .pseudo(.checked) ~ .star) > .star) % (
+    color(Colors.black)
+      <> fontWeight(.bold)
+    )
+    <> extraSpinnerStyles
+
+private func fetchSeatsTaken<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T3<Database.User, Int, A>, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, T2<Database.User, A>, Data> {
+
+    return { conn -> IO<Conn<ResponseEnded, Data>> in
+      let user = conn.data.first
+
+      let invitesAndTeammates = sequence([
+        parallel(AppEnvironment.current.database.fetchTeamInvites(user.id).run)
+          .map { $0.right?.count ?? 0 },
+        parallel(AppEnvironment.current.database.fetchSubscriptionTeammatesByOwnerId(user.id).run)
+          .map { $0.right?.count ?? 0 }
+        ])
+
+      return invitesAndTeammates
+        .sequential
+        .flatMap { middleware(conn.map(const(user .*. $0.reduce(1, +) .*. conn.data.second))) }
+    }
+}
+
+let subscriptionChangeShowView = View<(Stripe.Subscription, Database.User, Int)> { subscription, currentUser, seatsTaken -> Node in
+
+  gridRow([
+    gridColumn(sizes: [.mobile: 12, .desktop: 8], [style(margin(leftRight: .auto))], [
+      div(
+        [`class`([Class.padding([.mobile: [.all: 3], .desktop: [.all: 4]])])],
+        titleRowView.view(subscription)
+          <> formRowView.view((subscription, seatsTaken))
+          <> cancelRowView.view(subscription)
+      )
+      ])
+    ])
+}
+
+private let titleRowView = View<Stripe.Subscription> { subscription in
+  gridRow([`class`([Class.padding([.mobile: [.bottom: 2]])])], [
+    gridColumn(sizes: [.mobile: 12], [
+      div([
+        h1([`class`([Class.pf.type.responsiveTitle2])], ["Modify subscription"]),
+        p([
+          "You are currently enrolled in the ", strong([text(subscription.plan.name)]), " plan. ",
+          "Your subscription will ",
+          subscription.isRenewing ? "renew" : "end",
+          " on ",
+          strong([text(dateFormatter.string(from: subscription.currentPeriodEnd))]),
+          ".",
+          subscription.isRenewing
+            ? ""
+            : " Reactivate your subscription by submitting the form below."
```

Comment:
2 things:

1) When I tested locally to cancel my membership, I got a `We couldn’t cancel your subscription at this time.` error flash, but it did seem to actually cancel in stripe.

2) after canceling my account page becomes:

<img width="652" alt="screenshot 2018-03-07 08 41 24" src="https://user-images.githubusercontent.com/135203/37095257-55ee0824-21e3-11e8-843b-da2b14a63bd7.png">

And that `Reactivate` button does the reactivate action at once and sends me back to account. So, is this state in the change screen possible?

Reply:
1. Good catch. I'll try to repro figure out what's going on.
2. Yeah, I made the change to show the reactivation info in "change" just in case we wanted to change this. Hitting the change page is totally possible when you've canceled but it's still pending, but I didn't yet link to it. Thoughts?